workers/data_refinery_workers/processors/array_express.py

```diff
@@ -31,14 +37,32 @@ def cel_to_pcl(kwargs: Dict):
         kwargs["success"] = False
         return kwargs
 
-    temp_dir = file_management.get_temp_dir(batch)
+    input_file = file_management.get_temp_pre_path(batch)
     output_file = file_management.get_temp_post_path(batch)
 
-    ro.r('source("/home/user/r_processors/cel_to_pcl.R")')
-    ro.r['ProcessCelFiles'](
-        temp_dir,
-        "Hs",  # temporary until organism handling is more defined
-        output_file)
+    header = ro.r['::']('affyio', 'read.celfile.header')(input_file)
+
+    # header is a list of vectors. [0][0] contains the package name.
+    table = str.maketrans(dict.fromkeys(string.punctuation))
+    package_name = header[0][0].translate(table).lower()
+    brainarray_package = package_name + "hsentrezgprobe"
+
+    # Some CEL headers have a v1 in the package name which is not part
+    # of the brainararry package name. We have a table which corrects
+    # for this.
```

Comment:
We might need a better long term solution here -- it could work to strip anything after the `v` for version in most cases. Will think about it more.